crates/services/gas_price_service/src/common/fuel_core_storage_adapter.rs

```diff
@@ -67,20 +100,150 @@ where
             })?;
         Ok(metadata.map(|inner| inner.into_owned()))
     }
+}
 
+impl<Storage> GetDaSequenceNumber for StructuredStorage<Storage>
+where
+    Storage: KeyValueInspect<Column = GasPriceColumn> + Send + Sync,
+{
+    fn get_sequence_number(
+        &self,
+        block_height: &BlockHeight,
+    ) -> GasPriceResult<Option<u32>> {
+        let sequence_number = self
+            .storage::<SequenceNumberTable>()
+            .get(block_height)
+            .map_err(|err| GasPriceError::CouldNotFetchDARecord(err.into()))?
+            .map(|no| *no);
+        Ok(sequence_number)
+    }
+}
+
+pub struct NewStorageTransaction<'a, Storage> {
```

Comment:
We need this wrapper purely to dodge the orphan-rule, right?

Reply:
Exactly. Because the `UnrecordedBlocks` trait is in a separate crate. I think we could consider moving all the gas price algo stuff into the service, but until then we need a newtype.